smolyay/basis.py

```diff
@@ -72,6 +84,32 @@ def in_domain(self, x):
             numpy.greater_equal(x, self.domain[0]), numpy.less_equal(x, self.domain[1])
         )
 
+    def _scale_to_domain(self, points, old_domain):
+        """Scale points from a domain to BasisFunction domain
+
+        Parameters
+        ----------
+        points: numeric or array-like
+            points to be shifted to new domain.
+
+        old_domain: ndarray of shape (2,)
+            upper and lower bounds of points.
+
+        Returns
+        -------
+        numeric or array-like
+            points shifted to BasisFunction domain"""
+        points = numpy.array(points)
```

Comment:
This will always force a copy, right? Can we do this with a different NumPy method that won't force a copy if not needed?